config/environments/production.rb

```diff
@@ -17,42 +14,34 @@
   config.consider_all_requests_local       = false
   config.action_controller.perform_caching = false
 
-  # Attempt to read encrypted secrets from `config/secrets.yml.enc`.
-  # Requires an encryption key in `ENV["RAILS_MASTER_KEY"]` or
-  # `config/secrets.yml.key`.
-  config.read_encrypted_secrets = true
+  # Ensures that a master key has been made available in either ENV["RAILS_MASTER_KEY"]
+  # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
+  # config.require_master_key = true
 
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
   config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
-  config.serve_static_assets = true
-
-  if ENV["RAILS_LOG_TO_STDOUT"].present?
-    logger = ActiveSupport::Logger.new(STDOUT)
-    logger.formatter = config.log_formatter
-    config.logger = ActiveSupport::TaggedLogging.new(logger)
-  end
```

Comment:
I believe this lines are necessary for Heroku

Reply:
The `config.serve_static_assets = true` is old school and has been replaced by `config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?`. Heroku sets `RAILS_SERVE_STATIC_FILES` via it's ruby buildpack automatically.

Regarding `RAILS_LOG_TO_STDOUT`, yes you might be right. I'm not sure why Rails 6 didn't generate this... might be because it's handled deeper down the stack instead. But according to heroku docs looks like they're still expecting these lines to be present even in 6.x (https://devcenter.heroku.com/articles/getting-started-with-rails6#heroku-gems).

Would like to try this out on staging just to make sure. Rails team tends to work quite closely with Heroku to improve the out of the box experience with each version of rails so might be that this is no longer needed...

Reply:
Alright, I'll leave it to you to check and fix it if it doesn't.